packages/@o3r/eslint-plugin/src/rules/json/json-dependency-versions-harmonize/json-dependency-versions-harmonize.spec.ts

```diff
@@ -0,0 +1,134 @@
+import * as path from 'node:path';
+import * as fs from 'node:fs';
+import { TSESLint } from '@typescript-eslint/experimental-utils';
+import jsonDependencyVersionsHarmonize from './json-dependency-versions-harmonize';
+
+const ruleTester = new TSESLint.RuleTester({
+  parser: require.resolve('jsonc-eslint-parser'),
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: 'module'
+  }
+});
+
+const packageJsonWorkspace = {
+  workspaces: [
+    './packages/*'
+  ],
+  dependencies: {
+    myDep: '^2.0.0'
+  }
+};
+
+const packageJson2 = {
+  name: 'testPackage',
+  dependencies: {
+    myDep: '^1.0.0',
+    myOtherDep: '~2.0.0'
+  }
+};
+
+const packageToLint = path.resolve(__dirname, 'local', 'packages', 'my-package', 'package.json');
+
+beforeAll(() => {
+  fs.mkdirSync(path.resolve(__dirname, 'local'));
```

Comment:
it would probably be better to not rely on filesystem for the tests and mock `node:fs` module

Reply:
Actually I just copy the UT from another plugin test we have.
Unfortunately, mocking `node:fs` may not be enough because of `globby` and mocking both of them will make the test more unitary but less meaningful :S

Reply:
I'm not sure to understand your point, I think you do want the `fs` used by ESLint to be mocked as well so that it doesn't read from the disk

Reply:
`TSESLint` is using `fs` internally to manage different files loading (and using `buffer`, not `string` content...).
As we cannot mock `fs` only for our code with Jest, it's not possible to correctly mock `fs` to handle everything needed by `TSESlint`.